RobotConfig:
  robot_name: "hhfc"
  type: "gz"
  joint_size: 14
  motor_size: 14
  switching_time: 0.50
  joint_names: &joint_names
    - "l_hip_p"
    - "l_hip_r"
    - "l_hip_y"
    - "l_knee"
    - "l_ankle_p"
    - "l_ankle_r"
    - "r_hip_p"
    - "r_hip_r"
    - "r_hip_y"
    - "r_knee"
    - "r_ankle_p"
    - "r_ankle_r"
    - "l_shoulder_p"
    - "r_shoulder_p"
  motor_names: *joint_names

  motor_direction:
    l_hip_p: 1
    l_hip_r: 1
    l_hip_y: 1
    l_knee: 1
    l_ankle_p: 1
    l_ankle_r: 1
    r_hip_p: 1
    r_hip_r: 1
    r_hip_y: 1
    r_knee: 1
    r_ankle_p: 1
    r_ankle_r: 1
    l_shoulder_p: 1
    r_shoulder_p: 1

  observer:
    log_data: false # true
    log_dir: "/home/dknt/Project/bitbot_gz_ws/log/extra"
    motor_pos_filter:
      type: "PassThrough"
      lower_bound: [-3.14, -3.14, -3.14, -3.14, -3.14, -3.14,
                    -3.14, -3.14, -3.14, -3.14, -3.14, -3.14,
                    -3.14, -3.14]
      upper_bound: [3.14, 3.14, 3.14, 3.14, 3.14, 3.14,
                    3.14, 3.14, 3.14, 3.14, 3.14, 3.14,
                    3.14, 3.14] 
    motor_vel_filter:
      type: "Mean"
      history_length: 3
      lower_bound: [-50.0, -50.0, -50.0, -50.0, -50.0, -50.0,
                    -50.0, -50.0, -50.0, -50.0, -50.0, -50.0,
                    -50.0, -50.0]
      upper_bound: [50.0, 50.0, 50.0, 50.0, 50.0, 50.0,
                    50.0, 50.0, 50.0, 50.0, 50.0, 50.0,
                    50.0, 50.0] 
    ang_vel_filter:
      type: "PassThrough"
      upper_bound: [100.0, 100.0, 100.0]
      lower_bound: [-100.0, -100.0, -100.0]
    acc_filter:
      type: "PassThrough"
      upper_bound: [100.0, 100.0, 100.0]
      lower_bound: [-100.0, -100.0, -100.0]
    euler_filter:
      type: "PassThrough"
      upper_bound: [100.0, 100.0, 100.0]
      lower_bound: [-100.0, -100.0, -100.0]

    # Scan points parameters
    scan_size: 132
    scan_x_num: 12
    scan_y_num: 11
    scan_x_res: 0.15
    scan_y_res: 0.15
    scan_x_bias: -0.45
    scan_y_bias: -0.75

  executor:
    torque_limit:  # Nm
      l_hip_p: 200.0
      l_hip_r: 200.0
      l_hip_y: 200.0
      l_knee: 200.0
      l_ankle_p: 200.0
      l_ankle_r: 200.0
      r_hip_p: 200.0
      r_hip_r: 200.0
      r_hip_y: 200.0
      r_knee: 200.0
      r_ankle_p: 200.0
      r_ankle_r: 200.0
      l_shoulder_p: 200.0
      r_shoulder_p: 200.0
    motor_upper_limit:  # rad
      l_hip_p: 3.14
      l_hip_r: 3.14 
      l_hip_y: 3.14 
      l_knee: 3.14 
      l_ankle_p: 3.14 
      l_ankle_r:  3.14
      r_hip_p: 3.14
      r_hip_r: 3.14
      r_hip_y: 3.14
      r_knee: 3.14
      r_ankle_p: 3.14
      r_ankle_r: 3.14
      l_shoulder_p: 3.14
      r_shoulder_p: 3.14
    motor_lower_limit:  # rad
      l_hip_p: -3.14
      l_hip_r: -3.14
      l_hip_y: -3.14
      l_knee: -3.14
      l_ankle_p: -3.14
      l_ankle_r: -3.14
      r_hip_p: -3.14
      r_hip_r: -3.14
      r_hip_y: -3.14
      r_knee: -3.14
      r_ankle_p: -3.14
      r_ankle_r: -3.14
      l_shoulder_p: -3.14
      r_shoulder_p: -3.14

  init_pos:
    duration: 1.5
    init_position: &defult_init_pos
      l_hip_p: 0.3
      l_hip_r: 0.0
      l_hip_y: 0.0
      l_knee: -0.6
      l_ankle_p: -0.3
      l_ankle_r: 0.0
      r_hip_p: 0.3
      r_hip_r: 0.0
      r_hip_y: 0.0
      r_knee: -0.6
      r_ankle_p: -0.3
      r_ankle_r: 0.0
      l_shoulder_p: 0.0
      r_shoulder_p: 0.0

    p_gains: &default_p_gains
      l_hip_p: 150.0
      l_hip_r: 150.0
      l_hip_y: 100.0
      l_knee: 150.0
      l_ankle_p: 30.0
      l_ankle_r: 30.0
      r_hip_p: 150.0
      r_hip_r: 150.0
      r_hip_y: 100.0
      r_knee: 150.0
      r_ankle_p: 30.0
      r_ankle_r: 30.0
      l_shoulder_p: 100.0
      r_shoulder_p: 100.0

    d_gains: &default_d_gains
      l_hip_p: 1.5
      l_hip_r: 1.5
      l_hip_y: 1.0
      l_knee: 1.5
      l_ankle_p: 1.0
      l_ankle_r: 1.0
      r_hip_p: 1.5
      r_hip_r: 1.5
      r_hip_y: 1.0
      r_knee: 1.5
      r_ankle_p: 1.0
      r_ankle_r: 1.0
      l_shoulder_p: 1.0
      r_shoulder_p: 1.0

  zero_pos:
    duration: 1.0
    init_position: *defult_init_pos
    p_gains: *default_p_gains
    d_gains: *default_d_gains

  policy_standing:
    p_gains: *default_p_gains
    d_gains: *default_d_gains
    decimation: 10

    inference:
      policy_type: "HumanoidStand"
      use_absolute_clock: false
      control_period: 0.01
      stand_threshold: 0.045
      model_path: "/home/dknt/Project/bitbot_gz_ws/src/bitbot_gz/checkpoint/robust_beta.onnx"
      obs_buffer_size: 15

      log_data: false # true
      log_dir: "/home/dknt/Project/bitbot_gz_ws/log/policy"
      log_name: "standing"

      device: "CPU"
      cycle_time: 0.90
      single_obs_size: 47
      action_size: 12
      action_scale: 0.5
      obs_scales:
        ang_vel: 1.0
        lin_vel: 2.0
        command: 1.0
        dof_pos: 1.0
        dof_vel: 0.05
        proj_gravity: 1.0
      clip_action: 18.0
      realtime: true # false
      stick_to_core: 5

      policy_joint_names: &default_policy_joint_names
        - "l_hip_p"
        - "l_hip_r"
        - "l_hip_y"
        - "l_knee"
        - "l_ankle_p"
        - "l_ankle_r"
        - "r_hip_p"
        - "r_hip_r"
        - "r_hip_y"
        - "r_knee"
        - "r_ankle_p"
        - "r_ankle_r"

      policy_default_position: &default_policy_default_position
        l_hip_p: 0.3
        l_hip_r: 0.0
        l_hip_y: 0.0
        l_knee: -0.6
        l_ankle_p: -0.3
        l_ankle_r: 0.0
        r_hip_p: 0.3
        r_hip_r: 0.0
        r_hip_y: 0.0
        r_knee: -0.6
        r_ankle_p: -0.3
        r_ankle_r: 0.0

  policy_walking:
    p_gains: *default_p_gains
    d_gains: *default_d_gains
    decimation: 10

    inference:
      policy_type: "Humanoid"
      use_absolute_clock: false
      control_period: 0.01
      model_path: "/home/dknt/Project/bitbot_gz_ws/src/bitbot_gz/checkpoint/alpha_msl.onnx"
      obs_buffer_size: 15

      log_data: false #  true
      log_dir: "/home/dknt/Project/bitbot_gz_ws/log/policy"
      log_name: "walking"

      device: "CPU"
      cycle_time: 0.90
      single_obs_size: 47
      action_size: 12
      action_scale: 0.5
      obs_scales:
        ang_vel: 1.0
        lin_vel: 2.0
        command: 1.0
        dof_pos: 1.0
        dof_vel: 0.05
        proj_gravity: 1.0
      clip_action: 18.0
      realtime: true # false
      stick_to_core: 6

      policy_joint_names: *default_policy_joint_names
      policy_default_position: *default_policy_default_position

  policy_robust:
    p_gains: *default_p_gains
    d_gains: *default_d_gains
    decimation: 10

    inference:
      policy_type: "Humanoid"
      use_absolute_clock: false
      control_period: 0.01
      obs_buffer_size: 15
      model_path: "/home/dknt/Project/bitbot_gz_ws/src/bitbot_gz/checkpoint/terrain_alpha.onnx"

      log_data: false # true
      log_dir: "/home/dknt/Project/bitbot_gz_ws/log/policy"
      log_name: "robust"

      device: "CPU"
      cycle_time: 1.0
      single_obs_size: 47
      action_size: 12
      action_scale: 0.5
      obs_scales:
        ang_vel: 1.0
        lin_vel: 2.0
        command: 1.0
        dof_pos: 1.0
        dof_vel: 0.05
        proj_gravity: 1.0
      clip_action: 18.0
      realtime: true # false
      stick_to_core: 7

      policy_joint_names: *default_policy_joint_names
      policy_default_position: *default_policy_default_position

  policy_perceptive:
    p_gains: *default_p_gains
    d_gains: *default_d_gains
    decimation: 20
    perception_enabled: true

    inference:
      policy_type: "Perceptive"
      control_period: 0.02
      obs_buffer_size: 10
      # TODO
      model_path: "/home/dknt/Project/bitbot_gz_ws/src/bitbot_gz/checkpoint/terrain_alpha.onnx"

      log_data: false # true
      log_dir: "/home/dknt/Project/bitbot_gz_ws/log/policy"
      log_name: "robust"

      device: "CPU"
      cycle_time: 1.0
      single_obs_size: 45
      scan_size: 132
      action_size: 12
      action_scale: 0.5
      obs_scales:
        ang_vel: 1.0
        lin_vel: 2.0
        command: 1.0
        dof_pos: 1.0
        dof_vel: 0.05
        proj_gravity: 1.0
        scan: 1.0
      clip_action: 18.0
      realtime: false
      stick_to_core: 8

      policy_joint_names: *default_policy_joint_names
      policy_default_position: *default_policy_default_position
